examples/azure/terraform-azure-availabilityset-example/README.md

```diff
@@ -0,0 +1,109 @@
+it should be free, but you are completely responsible for all Azure charges.
+
+## Running this module manually
+
+1. Sign up for [Azure](https://azure.microsoft.com/).
+1. Configure your Azure credentials using one of the [supported methods for Azure CLI
+   tools](https://docs.microsoft.com/en-us/cli/azure/azure-cli-configuration?view=azure-cli-latest).
+1. Install [Terraform](https://www.terraform.io/) and make sure it's on your `PATH`.
+1. Run `terraform init`.
+1. Run `terraform apply`.
+1. When you're done, run `terraform destroy`.
+
+## Running automated tests against this module
+
+1. Sign up for [Azure](https://azure.microsoft.com/).
+1. Configure your Azure credentials using one of the [supported methods for Azure CLI
+   tools](https://docs.microsoft.com/en-us/cli/azure/azure-cli-configuration?view=azure-cli-latest).
+1. Install [Terraform](https://www.terraform.io/) and make sure it's on your `PATH`.
+1. [Review environment variables](#review-environment-variables).
+1. Install [Golang](https://golang.org/) and make sure this code is checked out into your `GOPATH`.
+1. `cd test`
+1. Make sure [the azure-sdk-for-go versions match](#check-go-dependencies) in [/go.mod](/go.mod) and in [test/azure/terraform_azure_availabilityset_example_test.go](/test/azure/terraform_azure_availabilityset_example_test.go).
+1. `go build terraform_azure_availabilityset_example_test.go`
+1. `go test -v -run TestTerraformAzureAvailabilitySetExample`
+
+## Module test APIs
+
+This module exposes the following top-level API methods:
+
+- `AvailabilitySetExists` indicates whether the speficied Azure Availability Set  exists\
+    func AvailabilitySetExists(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) bool
+
+- `GetAvailabilitySetFaultDomainCount` gets the Fault Domain Count for the specified Azure Availability Set\
+    func GetAvailabilitySetFaultDomainCount(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) int32
+
+- `CheckAvailabilitySetContainsVM` checks if the Virtual Machine is contained in the Availability Set VMs\
+    func CheckAvailabilitySetContainsVM(t testing.TestingT, vmName string, avsName string, resGroupName string, subscriptionID string) bool
+
+- `GetAvailabilitySetVMs` gets a list of VM names in the specified Azure Availability Set\
+    func GetAvailabilitySetVMs(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) []string
+    
+- `GetAvailabilitySetE` gets an Availability Set in the specified Azure Resource Group\
+    func GetAvailabilitySetE(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) (*compute.AvailabilitySet, error)
+    
+- `GetAvailabilitySetClientE` gets a new Availability Set client in the specified Azure Subscription\
+    func GetAvailabilitySetClientE(subscriptionID string) (*compute.AvailabilitySetsClient, error)
+
+## Check Go Dependencies
```

Comment:
I'm a bit confused why this is necessary?

Reply:
Was copied from a previous example, removed erroneous instruction.